core/trino-main/src/main/java/io/trino/operator/output/PositionsAppender.java

```diff
@@ -27,6 +27,12 @@
      */
     void appendRle(Block value, int rlePositionCount);
 
+    /**
```

Comment:
It feels like the cost of allocating single position arrays for every row might be significant. @lukasz-stec Do you know what most of the overhead comes from?

Reply:
Well, I could add an `if (positionCount == 1)` to appendRle to optimize it and then most of overhead would be that `if`. Not only during execution but also during compilation, which could impact batch append path.
IMO separate method is a lot cleaner and easier to optimize and since PositionsAppender is not a generic API to be used anywhere, at least not at the moment ,we shouldn't worry about adding a method to it.

Reply:
> it and then most of overhead would be that if

I don't think that would be an overhead. It won't be for batched append. And for row-by-row appends this `if` would evaluate to same value, so it's largely irrelevant.